src/ec/eddsa.rs

```diff
@@ -248,6 +250,104 @@ extern  {
     fn GFp_x25519_sc_reduce(s: &mut UnreducedScalar);
 }
 
+/// Calculate the modular inverse of field element |a| using Fermat's Little
+/// Theorem:
+///
+///    a**-1 (mod q) == a**(q - 2) (mod q)
+///
+/// The exponent (q - 2) is:
+///
+///    0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffeb
+///
+/// Note that we do not explicitly guard against the case |a| == 0, even though
+/// it has no multiplicative inverse.
+///
+/// This implementation of the modular exponentiation is ultimately derived
+/// from Daniel J. Bernstein's reference implementation in qhasm, via
+/// BoringSSL, and referenced in [Curve15519]. The overall strategy is to
+/// square and multiply against the fixed base |a| to double and add to the
+/// exponent, reusing intermediate computations as we are able.
+///
+/// [Curve25519]: https://cr.yp.to/ecdh.html#curve25519-paper
+#[doc(hidden)]
+#[no_mangle]
```

Comment:
In the end, this function should be renamed `elem_inverse`, it should have the type `Elem -> Elem` or `&Elem -> Elem`, it shouldn't be public, and it shouldn't be no_mangle.

(Aside: any `no_mangle` function in *ring* should have the ugly `GFp_` prefix. The ugliness is a feature.)